### DISCUSSION TITLE
Convert data to hetero using attribute

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#6069] [Q&A]
Hello! I'm trying to create heterogeneous data from homogeneous using to_heterogeneus like this:
```
nodeTypes = data.x[:, -1]
hetero_data = data.to_heterogeneous(nodeTypes)
```
Basically, the last attribute in my nodes is an integer (1 through 10 for example) that corresponds to the correct node group.
But, I get this:
`ValueError: Could not construct a 'HeteroData' object from the 'Data' object because single edge types span over multiple node types
`
I suppose there's something that I don't understand here.

**Comment by rusty1s** [comment]
This warning indicates that you have edge types in `edge_type` which reference different node types. This is currently not supported, but should probably be. Let me know if you wanna help fixing this.

**Reply by antoni4040** [reply]
So, every possible combination of node types need to be connected with a specific edge_type?

**Reply by rusty1s** [reply]
Yes, that is what we currently assume.

**Reply by antoni4040** [reply]
OK, let's assume 2 types of nodes, a and b. So, we need edge_types aa, ab and bb. Is there any easy way to convert from homogeneous to heterogeneous?

**Comment by rusty1s** [comment]
I guess you can do something like
```python
edge_type = num_node_types * edge_type + data.node_type[data.edge_index[0]]

**Reply by antoni4040** [reply]
I don't get it. :(

**Reply by rusty1s** [reply]
The code snippet I posted just ensures that edge types are not shared across different node types in the sense that we multiply the `edge_type` by the number of node types. For every `edge_type` that is shared across the same node type, we add its node type index to it.
